pkg/export/export.go

```diff
@@ -377,12 +392,6 @@ func (e *Exporter) triggerNext() {
 	}
 }
 
-// ClientName and Version are used to identify to User Agent. TODO(maxamin): automate versioning.
-const (
-	ClientName = "gmp-collector"
-	Version    = "v2.28.1-gmp.0"
-)
```

Comment:
Let's leave these in and not allow users to prevent them from showing up in the user agent. There's no good reason to override this. Looking at [common UA formatting](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent):

`Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0`

So our values should probably be

```
ClientName = "prometheus-engine-export"
Version = "0.1.0"
```

So that then with a binary-specific addition the UA would look like

```
Prometheus/2.28.1-gmp.1 prometheus-engine-export/0.1.0
rule-evaluator/0.1.0 prometheus-engine-export/0.1.0
```

The first part could be made overrideable via flags or envvar as you did here.

Reply:
Done. See my comments around why an env var was preferred.